src/domain/innovation-hub/innovation.hub.service.ts

```diff
@@ -73,11 +73,19 @@ export class InnovationHubService {
     hub.authorization = new AuthorizationPolicy();
     hub.listedInStore = true;
     hub.searchVisibility = SearchVisibility.ACCOUNT;
+    hub.account = account;
+
+    if (!account.storageAggregator) {
```

Comment:
i would fail sooner, no point of doing all that extra work if you already know it will fail.